repl/src/main/scala/ammonite/repl/interp/Pressy.scala

```diff
@@ -97,15 +104,56 @@ object Pressy {
-      index -> ask(index, pressy.askScopeCompletion).map(s =>
-        pressy.ask(() => (s.sym.name.decoded, None))
-      )
+      pressy.ask { () =>
+        def isNotPackage(target: pressy.Tree) = target.symbol == null || target.symbol == pressy.NoSymbol || !target.symbol.hasPackageFlag
+        
+        val suggestions = tree.collect { 
+          case q"$target.$op" if target.pos.end <= index && isNotPackage(target) => (target, op, None)
+          case q"$target.$op($argument)" if target.pos.end <= index && isNotPackage(target) => (target, op, Some(argument))
+        }.sortBy(index - _._1.pos.end).headOption.flatMap {
+          case (operationTarget, operator, argument) =>
+            val completionInvocation = new String(operationTarget.pos.source.content.slice(operationTarget.pos.start, operationTarget.pos.end))
+            val completionIncovationArguments = Seq('"' + operator.decoded + '"', argument.map(a => new String(a.pos.source.content.slice(a.pos.start, a.pos.end))).getOrElse("()")).mkString(", ")
+            val wrapperName = "SuggesterWrapper" + eval.getCurrentLine + suggestionNumber.incrementAndGet
+            val code = s"""$previousImports
+                        |object $wrapperName {
+                        |def suggestions = ($completionInvocation).suggestions($completionIncovationArguments)
+                        |}""".stripMargin
+//            println(s"($completionInvocation).suggestions($completionIncovationArguments)")
+            eval.evalClass(code, wrapperName) match {
+              case Res.Success((cls, imports)) =>
+                try {
+                  val suggestions = cls.getMethod("suggestions").invoke(null).asInstanceOf[Seq[String]]
+                  Some(index -> suggestions.map(s => s -> None).toList)
+                } catch { case ce: ClassCastException => None }
+              case _ => None
+            }
+        }.orElse {
+          tree.collect {
+            case q"StringContext(..$text).$op(..$args)" =>
+              val wrapperName = pressy.TermName("SuggesterWrapper" + eval.getCurrentLine + suggestionNumber.incrementAndGet)
+              val codeTree = q"""object $wrapperName { def suggestions = StringContext(..$text).$op.suggestions(..$args) }"""
+              val code = previousImports + "\n" + pressy.showCode(codeTree)
+//              println(code)
+              eval.evalClass(code, wrapperName.toString) match {
+                case Res.Success((cls, imports)) =>
+                  try {
+                    val suggestions = cls.getMethod("suggestions").invoke(null).asInstanceOf[Seq[String]]
+                    Some(index -> suggestions.map(s => s -> None).toList)
+                  } catch { case ce: ClassCastException => None }
+                case _ => None
+              }
+          }.headOption.flatten
+        }
+        suggestions getOrElse (index -> ask(index, pressy.askScopeCompletion).map(s => (s.sym.name.decoded, None)))
+      }
```

Comment:
I don't know if you are watching the diff in github with its very thin view, it does look good to me, and every line is a val declaration, except for the match and a try catch, there isn't any long chain of calls..
Anyway, I don't mind if someone wants to reformat it, I don't know the style you like so I wouldn't want to do it to avoid needless cycles of "make that look like that". Please reformat it to your heart content.